File: app/src/main/java/com/risalfajar/kamusoxford/Adapter/SearchResultAdapter.java
package com.risalfajar.kamusoxford.Adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.risalfajar.kamusoxford.Entity.SearchResult;
import com.risalfajar.kamusoxford.R;

public class SearchResultAdapter extends RecyclerView.Adapter<SearchResultAdapter.WordViewHolder> {

    private Context context;
    private SearchResult searchResult;

    public SearchResultAdapter(Context context) {
        this.context = context;
    }

    //region Getter and Setter
    public Context getContext() {
        return context;
    }

    public void setContext(Context context) {
        this.context = context;
    }

    public SearchResult getSearchResult() {
        return searchResult;
    }

    public void setSearchResult(SearchResult searchResult) {
        this.searchResult = searchResult;
    }

    //endregion

    @NonNull
    @Override
    public WordViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View itemRow = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_word, viewGroup, false);
        return new WordViewHolder(itemRow);
    }

    @Override
    public void onBindViewHolder(@NonNull WordViewHolder wordViewHolder, int i) {
        wordViewHolder.tvWord.setText(getSearchResult().getResults().get(i).getWord());
    }

    @Override
    public int getItemCount() {
        if(getSearchResult() != null)
            return getSearchResult().getResults().size();
        else
            return 0;
    }

    public class WordViewHolder extends RecyclerView.ViewHolder {

        TextView tvWord;

        public WordViewHolder(@NonNull View itemView) {
            super(itemView);
            tvWord = itemView.findViewById(R.id.tv_word);
        }
    }
}

File: app/src/main/java/com/risalfajar/kamusoxford/view/adapter/SearchResultAdapter.kt
package com.risalfajar.kamusoxford.view.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.risalfajar.kamusoxford.view.adapter.SearchResultAdapter.WordViewHolder
import com.risalfajar.kamusoxford.service.entity.SearchResult
import com.risalfajar.kamusoxford.R
import kotlinx.android.synthetic.main.item_word.view.*

class SearchResultAdapter : RecyclerView.Adapter<WordViewHolder>() {
    var searchResult: SearchResult? = null

    //endregion
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): WordViewHolder {
        val itemRow = LayoutInflater.from(viewGroup.context).inflate(R.layout.item_word, viewGroup, false)
        return WordViewHolder(itemRow)
    }

    override fun onBindViewHolder(wordViewHolder: WordViewHolder, i: Int) {
        wordViewHolder.tvWord.text = searchResult!!.results!![i].word
    }

    override fun getItemCount(): Int {
        return if (searchResult != null) searchResult!!.results!!.size else 0
    }

    inner class WordViewHolder(itemView: View) : ViewHolder(itemView) {
        val tvWord: TextView = itemView.tv_word
    }

}